grammar/abnf-grammar.txt

```diff
@@ -659,6 +658,19 @@ character-type = %s"char"
 
 scalar-type =  boolean-type / arithmetic-type / address-type / character-type
 
+; Circuit and Alias types are denoted by identifiers and the keyword `Self`.
+; The latter is only allowed inside a circuit definition,
+; to denote the circuit being defined.
+
+self-type = %s"Self"
+
+circuit-or-alias-type = identifier / self-type
```

Comment:
This has been replaced in more recent versions of the grammar, but perhaps it does not matter as this is a PR into the IR branch, not into master. We should double-check the "net diff" when we have a PR into master.

Reply:
this PR will be going into master once IR is merged. it was just pointed at IR for now to make sure that there are not conflicts between our branches while we wait for IR to be merged